python/lsst/analysis/ap/apdb.py

```diff
@@ -81,21 +145,33 @@ def load_sources_for_object(self, dia_object_id, limit=100000):
         data : `pandas.DataFrame`
             DiaSources for the specified diaObject.
         """
+
+        where_clauses = [f'"DiaSource"."diaObjectId" = {dia_object_id}']
+
+        if exclude_flagged:
+            where_clauses.append(self._make_flag_exclusion_clause(self.bad_DiaSource_flags))
+
+        where = "WHERE " + " and ".join(where_clauses) if len(where_clauses) else ""
+
+        order = 'ORDER BY "ccdVisitId", "diaSourceId"'
+        limit_str = f"LIMIT {limit}" if limit is not None else ""
+        query = ('SELECT * FROM "DiaSource"'
+                 f' {where} {order} {limit_str};')
```

Comment:
Good call pulling these all out of the connection context manager. Mind doing that on a separate commit for the rest of the `load_` methods?